src/Analyzer.Core/TemplateAnalyzer.cs

```diff
@@ -100,8 +102,58 @@ public IEnumerable<IEvaluation> AnalyzeTemplate(string template, string template
                 }
             }
 
+            var templateContext = new TemplateContext
+            {
+                OriginalTemplate = null,
+                ExpandedTemplate = null,
+                IsMainTemplate = true,
+                ResourceMappings = null,
+                TemplateIdentifier = templateFilePath,
+                IsBicep = isBicep,
+                SourceMap = sourceMap,
+                PathPrefix = "",
+                ParentContext = null
+            };
+            var  evaluations = AnalyzeAllIncludedTemplates(template, parameters, templateFilePath, templateContext, "");
+
+            // For each rule we don't want to report the same line more than once
+            // This is a temporal fix
+            var evalsToValidate = new List<IEvaluation>();
+            var evalsToNotValidate = new List<IEvaluation>();
+            foreach (var eval in evaluations)
+            {
+                if (!eval.Passed && eval.Result != null)
+                {
+                    evalsToValidate.Add(eval);
+                }
+                else
+                {
+                    evalsToNotValidate.Add(eval);
+                }
+            }
+            var uniqueResults = new Dictionary<(string, int), IEvaluation>();
+            foreach (var eval in evalsToValidate)
+            {
+                uniqueResults.TryAdd((eval.RuleId, eval.Result.LineNumber), eval);
+            }
+            evaluations = uniqueResults.Values.Concat(evalsToNotValidate);
+
+            return evaluations;
+        }
+
+        /// <summary>
+        /// Analyzes ARM templates, recursively going through the nested templates
+        /// </summary>
+        /// <param name="populatedTemplate">The ARM Template JSON with inherited parameters, variables, and functions, if applicable</param>
+        /// <param name="parameters">The parameters for the ARM Template JSON</param>
+        /// <param name="templateFilePath">The ARM Template file path</param>
+        /// <param name="parentContext">Template context for the immediate parent template</param>
+        /// <param name="pathPrefix"> path prefix</param>
```

Comment:
Use a better description here.

Reply:
lol I wrote that when initially implementing it and just forgot to update it